### PR TITLE
Generic Ranged Weapons

### PR DESCRIPTION
**Pull request by MemeWizard**
#### Changelog:
- Added generic ranged artifacts

#### How to test:

- Check that artifact stats are correct. Use [this](https://docs.google.com/spreadsheets/d/1fPiP2HnYhC80hOelgUdeuvItH__uX5H27NFSOwH-tyA/edit#gid=1787561420&range=48:50) and [this](https://docs.google.com/spreadsheets/d/1xDjRHfG-ovKVcEQVfv02BsK3j0dgwS2rfH7KJ4GirA0/edit#gid=1700305504&range=5:54) spreadsheets.
- Proofread [this](https://docs.google.com/spreadsheets/d/1FVVfKRCmTzPfiqoKvcfllqOhR613hpCzm2XJ8Gk1moU/edit#gid=15614703&range=1:91)
- Check that names are assigned correctly. Don't forget that there is 2 names variation - `[№1] [Type] of [№3] [№4]` (e.g. `Marksman's Bow of Wisp Vanquishing`) and `[Type] of [№2] [№3] [№4]` (e.g. `Rifle of Scary Octopus Repelling`)

### CODE REVIEW

**Comment by MemeWizard** [comment]
@Myrran thanks for review!

> Bow, crossbow and rifle artifacts doesn’t specify class requirements as other weapons do.

I wanted to ask other devs which classes are supposed to wield ranged weapons, but I kinda forgot about it and didn't fill their active trigger with any class conditions. *Sorry!*
Thanks for reminder, I'll change that later.

> Loc change suggestions:

I'll leave it to English Watchers, since it's not my native language. Thank you, anyways.

>Two-handed maces level 2-5 have support troop bonus, while according to specs they should have shock bonus.

True, thanks for noticing! I'll add this to my do-to list.

**Comment by MemeWizard** [comment]
> Bow, crossbow and rifle artifacts doesn’t specify class requirements as other weapons do.

@Myrran, I changed their requirements. For now, they are identical to melee weapons, but I want to change that later in other PR.

> I agree with all the suggestions he made, and I think he caught them all.

Implemented it, then

File: common/scripted_effects/wc_artifacts_generic_scripted_effects.txt [hunk 1]
```diff
@@ -750,56 +880,102 @@ game_start_spawn_artifacts_random_effect = { # Slots -> Types
...
+						5 = { add_artifact = halberd_1 }
+						# Spears
+						1 = { add_artifact = spear_5 }
+						2 = { add_artifact = spear_4 }
+						3 = { add_artifact = spear_3 }
+						4 = { add_artifact = spear_2 }
+						5 = { add_artifact = spear_1 }
+					}
+				}
+				3 = { # Ranged Weapons
```
Comment (zumbak04) [inline common/scripted_effects/wc_artifacts_generic_scripted_effects.txt:943]: Wait, the ranged weapons have more chances to spawn than the melee ones?

File: interface/wc_treasury_generic.gfx [hunk 2]
```diff
@@ -2131,4 +2131,151 @@ spriteTypes = {
...
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\crossbow_1.dds"
+	}
+	spriteType = {
+		name = "GFX_spear_5"
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\spear_5.dds"
+	}
+	spriteType = {
+		name = "GFX_spear_5_small"
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\spear_5.dds"
+	}
```
Comment (zumbak04) [inline interface/wc_treasury_generic.gfx:2231]: Duplicated sprites.
```suggestion
```

File: interface/wc_treasury_generic.gfx [hunk 3]
```diff
@@ -2131,4 +2131,151 @@ spriteTypes = {
...
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\bow_1.dds"
+	}
+	spriteType = {
+		name = "GFX_spear_5"
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\spear_5.dds"
+	}
+	spriteType = {
+		name = "GFX_spear_5_small"
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\spear_5.dds"
+	}
```
Comment (zumbak04) [inline interface/wc_treasury_generic.gfx:2182]: Duplicated sprites.
```suggestion
```

File: interface/wc_treasury_generic.gfx [hunk 4]
```diff
@@ -2131,4 +2131,151 @@ spriteTypes = {
...
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\rifle_1.dds"
+	}
+	spriteType = {
+		name = "GFX_spear_5"
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\spear_5.dds"
+	}
+	spriteType = {
+		name = "GFX_spear_5_small"
+		texturefile = "gfx\\interface\\inventory\\artifacts\\weapons_generic\\spear_5.dds"
+	}
```
Comment (zumbak04) [inline interface/wc_treasury_generic.gfx:2280]: Duplicated sprites.
```suggestion
```

File: localisation/customizable_localisation/wc_customizable_localisation.txt [hunk 5]
```diff
@@ -1306,19 +1306,25 @@ defined_text = {
...
+		# trigger = {
+			# is_hunter_class_trigger = yes
+			# prefers_guns_trigger = yes
+		# }
+		# localisation_key = GetClassSkill_the_gun
+	# }
+	# text = {
+		# trigger = {
+			# is_hunter_class_trigger = yes
+			# prefers_guns_trigger = no
```
Comment (zumbak04) [inline localisation/customizable_localisation/wc_customizable_localisation.txt:1319]: But what about the trigger? It's not used anywhere except this place.

File: common/scripted_effects/wc_artifacts_generic_scripted_effects.txt (hunk 1 above)
Comment (MemeWizard) [reply common/scripted_effects/wc_artifacts_generic_scripted_effects.txt:943]: Changed

File: localisation/customizable_localisation/wc_customizable_localisation.txt (hunk 5 above)
Comment (MemeWizard) [reply localisation/customizable_localisation/wc_customizable_localisation.txt:1319]: I want to use that trigger later

File: interface/wc_treasury_generic.gfx (hunk 2 above)
Comment (MemeWizard) [reply interface/wc_treasury_generic.gfx]: Thanks for noticing <3

File: common/scripted_effects/wc_artifacts_generic_scripted_effects.txt [hunk 6]
```diff
@@ -750,56 +880,102 @@ game_start_spawn_artifacts_random_effect = { # Slots -> Types
...
+
+							random_list = {
+								1 = { add_artifact = crossbow_5 }
+								2 = { add_artifact = crossbow_4 }
+								3 = { add_artifact = crossbow_3 }
+								4 = { add_artifact = crossbow_2 }
+								5 = { add_artifact = crossbow_1 }
+							}
+						}
+						1 = { # Rifles
```
Comment (zumbak04) [inline common/scripted_effects/wc_artifacts_generic_scripted_effects.txt:966]: Just an idea. Shouldn't crossbows and rifles be more common than bows in the societies with high techologies?

Comment (MemeWizard) [reply common/scripted_effects/wc_artifacts_generic_scripted_effects.txt:966]: Done

**Comment by MemeWizard** [comment]
@Myrran @zumbak04 @LordSilvermane <3

**Pull request merged**